merk/src/tree/kv.rs

```diff
@@ -15,12 +16,13 @@ use crate::tree::{
 //       field and value field.
 
 /// Contains a key/value pair, and the hash of the key/value pair.
-#[derive(Clone)]
+#[derive(Clone, Debug)]
 pub struct KV {
     pub(super) key: Vec<u8>,
     pub(super) value: Vec<u8>,
-    pub(super) hash: Hash,
-    pub(super) value_hash: Hash,
+    pub(crate) value_defined_cost: Option<u32>,
```

Comment:
Noticed we don't encode the value_defined_cost for KV, and we decode it to None, is this by design?

Reply:
It doesn't matter as this is only used during insert... maybe there's a way to get rid of it. I'll add a todo.